binding/form_mapping.go

```diff
@@ -57,11 +28,25 @@ func mapForm(ptr interface{}, form map[string][]string) error {
 var emptyField = reflect.StructField{}
 
 func mapFormByTag(ptr interface{}, form map[string][]string, tag string) error {
-	_, err := mapping(reflect.ValueOf(ptr), emptyField, form, tag)
+	return mappingByPtr(ptr, formSource(form), tag)
+}
+
+type setter interface {
+	Set(value reflect.Value, field reflect.StructField, key string, opt setOptions) (isSetted bool, err error)
+}
+
+type formSource map[string][]string
+
+func (form formSource) Set(value reflect.Value, field reflect.StructField, tagValue string, opt setOptions) (isSetted bool, err error) {
```

Comment:
check: `var _ setter = &formSource`
and add node for export method/class/interface.